### DISCUSSION TITLE
Best way to structure a site that uses a lot of Fetches and also how to refresh background content from a modal

### DISCUSSION THREAD

**Discussion by devforbetter** [vercel/next.js#60218] [Help]
### Summary

So I am learning Next.js 14 and the App Router (using seperate client and server components). The App i'm building relys on a lot of fetches to Vercels Postrgres. So the way I have it structured currently is:

Page.js contains the main / inital fetches of data, then passes the data to the child components and so on. What I need to get my head round is if the initial data is pulled in the Page.js and then I pass that to say a popup Modal component and then maybe further to a list of child components within that Modal. If I make a change in that modal, like inserting a new record into db. How can I then refresh the page (list of records) behind that modal when it is a completely different component? 

I tried using revalidatePath but that just gives me an error:

Uncaught (in promise) Error: Invariant: static generation store missing in revalidateTag _N_T_/sites

Someone then suggested move that into a separate actions.js (im not using TS), which then resulted in an error in the console about Fetch Failed POST or something.

So im at a bit of a loss on what I should be doing and the correct way to structure things. 

Page.js contains 2 components SiteList and AddSiteModal:
-----> SiteList -----> SiteListItems
-----> AddSiteModal -----> AddSiteModalListItem

So i'm trying to trigger the SiteList to re-fetch (which initially happens in page.js) when a button is click within the AddSiteModalListItem component.

**Comment by don-esteban** [comment]
Not sure, if I understand your issue, but it seems like you need to resolve this on React level with the help of Hooks. Combine `useState` or `useReducer` together with `useContext` for state management and to share state between components.

The current React docs are very good. As a starting point, I suggest reading [Managing State](https://react.dev/learn/managing-state)
